settlement/dymension/dymension.go

```diff
@@ -508,11 +551,11 @@ func (d *HubClient) waitForBatchInclusion(batchStartHeight uint64) (*settlement.
 		if err != nil {
 			return err
 		}
-		if latestBatch.Batch.StartHeight == batchStartHeight {
-			resultRetriveBatch = latestBatch
-			return nil
+		if latestBatch.Batch.StartHeight != batchStartHeight {
+			return settlement.ErrBatchNotFound
 		}
-		return settlement.ErrBatchNotFound
+		resultRetriveBatch = latestBatch
+		return nil
 	}, retry.Context(d.ctx), retry.LastErrorOnly(true),
 		retry.Delay(d.batchRetryDelay), retry.Attempts(d.batchRetryAttempts), retry.MaxDelay(batchRetryMaxDelay))
```

Comment:
this code is duplicated a lot
can we dry out by extracting d.retry(fn) which uses these default args?